File: 实验软件客户端/CPP源码/myinformationdialog.cpp
#include "myinformationdialog.h"
#include "ui_myinformationdialog.h"
#include <QDebug>
#include <QString>
#include <QMessageBox>
#include <fstream>
#include <io.h>
myinformationdialog::myinformationdialog(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::myinformationdialog)
{
    ui->setupUi(this);
    infopath = ".//information";
    iniSignalSlots();
}

void myinformationdialog::iniSignalSlots()
{
    connect(ui->btnYes,SIGNAL(pressed()),this,SLOT(GetAllInformation()));
    connect(this,SIGNAL(close_event(QString)),this,SLOT(close()));
}

void myinformationdialog::GetAllInformation()
{
    if (ui->nameEdit->text() == "")
     {
        QMessageBox::warning(this,"信息框","未填写姓名");
        return;
     }
    if (ui->ageEdit->text() == "")
     {
        QMessageBox::information(this,"信息框","未填写年龄");
        return;
     }
    if (ui->numberEdit->text() == "")
     {
        QMessageBox::information(this,"信息框","未填写编号");
        return;
     }
    if (ui->majorEdit->text() == "")
     {
        QMessageBox::information(this,"信息框","未填写专业");
        return;
    }
    auto result = QMessageBox::question(this,"信息框","信息填写已完成，是否确认",QMessageBox::Yes|
                                  QMessageBox::Cancel);
    if (result == QMessageBox::Yes)
    {
        QString data_path = infopath+"//"+ui->numberEdit->text()+".txt";
        if(!access(data_path.toUtf8().data(), 0))
        {
            auto result2 = QMessageBox::question(this,"信息框","信息已存在，是否覆盖",QMessageBox::Yes|
                                          QMessageBox::Cancel);
            if(result2== QMessageBox::Cancel)
            {
               QMessageBox::information(this,"信息框","你可以重新填写信息");
               return;
            }
        }
        std::ofstream ofs;
        ofs.open(data_path.toUtf8().data());
        if(!ofs.is_open())
        {
            QMessageBox::information(this,"信息框","文件路径打开失败");
            return;
        }
        ofs << ui->nameEdit->text().toUtf8().data() <<" "<<ui->ageEdit->text().toUtf8().data() << " "
            << ui->numberEdit->text().toUtf8().data() <<" "<<ui->majorEdit->text().toUtf8().data();
        if (ui->manButton->isChecked())
            ofs <<" 男";
        else
            ofs <<" 女";
        ofs.close();
    }
    emit close_event(ui->numberEdit->text());
}
myinformationdialog::~myinformationdialog()
{
    delete ui;
}

File: 实验软件客户端/IPOID_eyetrack_pyqt/myInformationDialog.py
import sys

import os

from PyQt5.QtWidgets import QApplication,QDialog,QMessageBox,QFileDialog

from PyQt5.QtCore import pyqtSlot,pyqtSignal

from ui_InformationDialog import Ui_InformationDialog

class QmyInformationDialog(QDialog):
    beginTest = pyqtSignal(str)
    def __init__(self,parent=None):
        super().__init__(parent)
        self.ui = Ui_InformationDialog()
        self.ui.setupUi(self)
        self.list = []
        self.inforpath ='./AllData'

    def on_btnYes_pressed(self):
        if (self.ui.nameEdit.text() == ""):
            dlgTitle ="信息框"
            strInfo ="未填写姓名"
            QMessageBox.information(self,dlgTitle,strInfo)
            return
        if (self.ui.ageEdit.text() == ""):
            dlgTitle ="信息框"
            strInfo ="未填写年龄"
            QMessageBox.information(self,dlgTitle,strInfo)
            return
        if (self.ui.numberEdit.text() == ""):
            dlgTitle ="信息框"
            strInfo ="未填写编号"
            QMessageBox.information(self,dlgTitle,strInfo)
            return
        if (self.ui.majorEdit.text() == ""):
            dlgTitle ="信息框"
            strInfo ="未填写专业"
            QMessageBox.information(self,dlgTitle,strInfo)
            return
        dlgTitle ="信息框"
        strInfo ="信息填写已完成，是否确认"
        result = QMessageBox.question(self,dlgTitle,strInfo,QMessageBox.Yes|
                                      QMessageBox.Cancel)
        if (result == QMessageBox.Yes):
            self.list = []
            self.list.append(str(self.ui.nameEdit.text()))
            self.list.append(str(self.ui.ageEdit.text()))
            self.list.append(str(self.ui.numberEdit.text()))
            self.list.append(str(self.ui.majorEdit.text()))
            if (self.ui.manButton.isChecked()):
                self.list.append("男")
            else:
                self.list.append("女")
             
            data_path = self.inforpath+'/'+self.ui.numberEdit.text()
            isExists=os.path.exists(data_path)
            if not isExists:
                 os.makedirs(data_path)
                 f = open('%s/information.txt'%(data_path),mode='w')
                 sep = ","
                 f.write(sep.join(self.list))
                 f.close()
                 self.beginTest.emit(data_path)
                 self.close()
            else:
                 dlgTitle ="信息框"
                 strInfo ="该编号已经存在，是否覆盖"
                 result_new = QMessageBox.question(self,dlgTitle,strInfo,QMessageBox.Yes|QMessageBox.Cancel)
                 if (result_new == QMessageBox.Yes):     
                     f = open('%s/information.txt'%(data_path),mode='w')
                     f.truncate(0)
                     sep = ","
                     f.write(sep.join(self.list))
                     f.close()
                     self.beginTest.emit(data_path)
                     self.close()
                 else:
                     dlgTitle ="信息框"
                     strInfo ="你可以重新填写信息"
                     QMessageBox.information(self,dlgTitle,strInfo)
                     return            
        if (result == QMessageBox.Cancel):
            dlgTitle ="信息框"
            strInfo ="你可以重新填写信息"
            QMessageBox.information(self,dlgTitle,strInfo)
            return

    @pyqtSlot(bool)
    def beginInformation(self,begin):
        self.show()

    
    def on_btnSetInfoPath_pressed(self):
        dirStr = QFileDialog.getExistingDirectory()
        if (dirStr == ""):
            return
        self.ui.InfoPathEdit.setText(dirStr)




if __name__ == "__main__":
    app = QApplication(sys.argv)
    form = QmyInformationDialog()
    form.show()
    sys.exit(app.exec_())
